app/controllers/submissions_controller.rb

```diff
@@ -560,8 +565,10 @@ def download_file_zip
   def download
     preview = params[:preview] == 'true'
     nbconvert_enabled = Rails.application.config.nbconvert_enabled
-
-    if FileHelper.get_file_type(params[:file_name]) == 'jupyter-notebook' && preview && nbconvert_enabled
+    rmd_convert_enabled = Rails.application.config.rmd_convert_enabled
+    file_type = FileHelper.get_file_type(params[:file_name])
+    if ((file_type == 'jupyter-notebook' && nbconvert_enabled) \
+     || (file_type == 'markdown' && rmd_convert_enabled)) && preview
```

Comment:
Ah, this will also affect `.md` files, not just `.Rmd` files. Modify `FileHelper.get_file_type` to return a different file type for `.Rmd` files to ensure that they aren't affected by this change

Reply:
@soheegoo sorry one thing I forgot to mention about this: we are using a syntax-highlighting library in the front-end for plaintext files. Please double-check the syntax highlighting behaviour for RMarkdown files when the `rmd_convert` option is *false*, as we still want the syntax highlighting to show up correctly in that case. Thanks!

Reply:
@david-yz-liu Should I set the syntax highlighting css classname to 'language-r' or 'language-markdown' (previous behaviour before this PR)? They highlight in different ways. The highlighting doesn't show up right now because PrismJS doesn't recognize 'language-rmarkdown', after I changed the FileHelper.get_file_type to return a different type for .Rmd files.

Reply:
@soheegoo great question, yes you could definitely have a special case _in the front-end_ to modify the file type when setting the css classname

Reply:
@david-yz-liu Thanks, I will do that. I was wondering if you have a preference on if I should set the syntax highlighting style to 'language-r' or 'language-markdown'?

Reply:
markdown is more accurate